zerver/lib/exceptions.py

```diff
@@ -473,3 +474,25 @@ class ValidationFailureError(JsonableError):
     def __init__(self, error: ValidationError) -> None:
         super().__init__(error.messages[0])
         self.errors = error.message_dict
+
+
+class MessageMoveError(JsonableError):
+    code = ErrorCode.MESSAGE_MOVE_TIME_LIMIT_RESTRICTION
+    data_fields = [
+        "total_messages_in_topic",
+        "messages_allowed_to_move",
+    ]
+
+    def __init__(
+        self,
+        total_messages_in_topic: int,
+        messages_allowed_to_move: int,
+    ) -> None:
+        self.total_messages_in_topic = total_messages_in_topic
+        self.messages_allowed_to_move = messages_allowed_to_move
+
+    @staticmethod
+    def msg_format() -> str:
+        return _(
+            "Can't move all messages in this topic. You can only move {messages_allowed_to_move}/{total_messages_in_topic} latest messages"
+        )
```

Comment:
I think I like the idea of this explaining which restriction (stream move or topic move) is the issue. Also this should mention "permission". For example: "You do not have permission to move messages older than {date} between streams. Onl {messages_allowed_to_move}/{total_messages_in_topic} will be moved."

Reply:
Are we fine with showing date when the time limit is set to some minutes or hours. We can though mention that error is due to time limit to improve the message.

Also, there can be a case when both stream and topic move is requested and both are not allowed for some messages and we move the messages for which both are allowed. So, the number of messages not allowed to do stream and topic edit might be different and also the date after which messages are allowed to move would be different. Do, we need to mention both of them in such case?

Reply:
Yeah let's go with a bit simpler. Perhaps `You only have permission to move the most recent messages in this topic ({total_messages_allowed_to_move}/{total_messages_in_topic}).`